### ISSUE TITLE
"Do you really want to update all rows" on statement with where clause

### ISSUE THREAD

**Issue by trondhindenes** [piccolo-orm/piccolo#444]
I'm preparing to run some updates as part of a migration, here's my test script that I'll eventually add to the migration:

We're changing the `onboarding_completed`flag to be `Boolean(default=False, null=False)` and plan to run this script beforehand:
```
import anyio
from piccolo.apps.migrations.auto.migration_manager import MigrationManager
from piccolo.columns.column_types import Boolean
from piccolo.table import Table
import asyncer


class Customer(Table):
    onboarding_completed: bool = Boolean(default=None, null=True)


ID = "2022-03-01T11:29:21:677517"
VERSION = "0.69.1"
DESCRIPTION = ""


async def set_all_onboarding_completed_to_false():
    res = await Customer.update({Customer.onboarding_completed: False}).where(Customer.onboarding_completed is None)
    print(res)

anyio.run(set_all_onboarding_completed_to_false)
```

However, this triggers the error
```
Do you really want to update all rows in Customer? If so, use pass `force=True` into `Customer.update`. Otherwise, add a where clause.
```
Which I'm finding strange, since there is a where clause here.

**Comment by dantownsend** [comment]
@trondhindenes I think this is the problem: `.where(Customer.onboarding_completed is None)`.

Try this: `.where(Customer.onboarding_completed.is_null())`

**Comment by trondhindenes** [comment]
yeah, I just saw it. Changing to "== True" also works. Hm, it would be awesome if piccolo somehow could warn that `is True` is an incorrect where clause. But, the quick is documented, it was just me not paying attention. So this isn't really a bug.

**Comment by dantownsend** [comment]
Yes, you're right - Piccolo could output a warning in this situation. It just needs to check if a `bool` is being passed in, as this is probably an accident.

In an ideal world we could just override the `is` opeator, and `SomeTable.some_column is None` would work, but I don't think it's possible currently with Python.

**Comment by dantownsend** [comment]
I've opened a PR to fix this. Piccolo will now raise a `ValueError` if a boolean is accidentally passed into `where` clause.

**Comment by trondhindenes** [comment]
hm that will affect https://github.com/piccolo-orm/piccolo/issues/436 won't it? at least pycharm tells me that all my where clauses are booleans.
I don't know if there's some smart way to determine if `is` was used instead of `==`?

**Comment by dantownsend** [comment]
I don't think there's a way to detect if `is` was used.

I think Pycharm mistakenly thinks `MyTable.some_column = 'abc'` returns a bool, but we override `__eq__` so it returns a `Where`:

https://github.com/piccolo-orm/piccolo/blob/5aea0fdc4f40a99615f534ec49a38b242a5f5cb0/piccolo/columns/base.py#L664

It could be a limitation of Pycharm's proprietary type checker. I'd raise an issue to ask them, but I'm not sure where to do it.

For `Boolean` columns, we actually have an `eq` method:

https://github.com/piccolo-orm/piccolo/blob/5aea0fdc4f40a99615f534ec49a38b242a5f5cb0/piccolo/columns/column_types.py#L1091

This means we can do `MyTable.some_boolean_column.eq(True)`. We could move this method, so it can be used on any column type. It could be a good solution.

**Comment by dantownsend** [comment]
@trondhindenes  Interesting - thanks for checking that.

So it does seem to be a Piccolo issue then.

I just noticed that there's a `# type: ignore` comment next to the `__eq__` method:

https://github.com/piccolo-orm/piccolo/blob/5aea0fdc4f40a99615f534ec49a38b242a5f5cb0/piccolo/columns/base.py#L664

Perhaps this is indicating to Pycharm to ignore the type hints, which is why it thinks a boolean should be returned.

I can't remember why `# type: ignore` was even required - will see if removing it fixes the issue.

**Comment by dantownsend** [comment]
I tried removing `type: ignore`, and it made no difference.

In my fresh install of Pycharm CE, I did a little sample project.

It seems to correctly realise that `Band.name == 'Band 1'` returns a `Where`.

<img width="686" alt="Screenshot 2022-03-03 at 08 27 32" src="https://user-images.githubusercontent.com/350976/156574557-6bf91936-6d32-45da-b38a-b3a330d603f4.png">

But thinks that `Band.manager.name ==  'Guido'` is a `bool`: 

<img width="686" alt="Screenshot 2022-03-03 at 08 27 43" src="https://user-images.githubusercontent.com/350976/156574544-abb212ba-6f98-4a64-b4ae-f410c2c8327b.png">

If `Band.name == 'Band 1'` returns a bool for you, perhaps that's the Pydantic plugin being a nuisance?
